hackage-upload/action.yml

```diff
@@ -0,0 +1,27 @@
+name: Upload to Hackage
+description: Upload an sdist archive to hackage.haskell.org
+
+inputs:
+  archive:
+    description: The sdist archive to upload
+    required: true
+  token:
+    description: The Hackage auth token
+    required: true
+  candidate:
+    description: Whether to upload the package as a candidate
+    default: true
+
+runs:
+  using: composite
+  steps:
+    - run: |
+        VENV=${GITHUB_ACTION_PATH}/.venv
```

Comment:
This is going to be an odd request, but given how small the action is, is it possible to have it written using javascript so that no extra dependencies need to be installed on the runner? Installing a python3 VENV seems very heavy weight for what is ultimately `curl -XPOST ...`

In theory, it shouldn't need any dependencies, although in practice it should use the actions/core package at the very least for reading and parsing inputs.

Reply:
Not odd at all! I originally tried using `curl` instead, but it seemed much more complicated than using `requests`... but apparently it's just a `-F 'package=@archive.tar.gz'`, so I'll switch to `curl`. I also switched to Javascript for the input parsing, using the `actions/github-script` action. I like this better than committing `node_modules` to the repo, as it makes releasing trivial